### PR TITLE
[Caching] Add CAS size management APIs

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
rdar://121944849

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
Depends on https://github.com/apple/swift/pull/71257

File: Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift [hunk 1]
```diff
@@ -198,6 +198,32 @@ public final class SwiftScanCAS {
     return try scanner.toSwiftString(casid)
   }
 
+  public var supportsSizeManagement: Bool {
+    scanner.supportsCASSizeManagement
+  }
+
+  public func getStorageSize() throws -> Int64 {
```
Comment (akyrtzi) [inline Sources/SwiftDriver/SwiftScan/SwiftScanCAS.swift:205]: Better to make it optional and return `nil` if "not supported". That way the caller has the option to handle "not supported" however they want, e.g. ignore it, emit only a warning, etc.

**Review by akyrtzi — APPROVED** [review]
Just a suggestion for a change, otherwise LGTM.

**Comment by cachemeifyoucan** [comment]
@swift-ci please test

**Comment by cachemeifyoucan** [comment]
@swift-ci please test windows platform

**Pull request merged**